packages/website/blog/2024-11-30-catching-any.mdx

```diff
@@ -0,0 +1,237 @@
+
+### Banning `Function`
+
+[`@typescript-eslint/no-unsafe-function-type`](https://typescript-eslint.io/rules/no-unsafe-function-type) reports on any instance of the built-in `Function` type written in source code.
+`Function` is a loose, unsafe type: it allows being called with any number of arguments and returns type `any`.
+
+Take the following version of `greet` that takes in a function for its parameter.
+`Function` doesn't describe any parameter or return types, so TypeScript can't know what types it's meant to take in or return.
+`@typescript-eslint/no-unsafe-function-type` would report on that `Function`:
+
+```ts
+function greet(getFriend: Function) {
+  //                      ~~~~~~~~
+  // eslint(@typescript-eslint/no-unsafe-function-type):
+  // The `Function` type accepts any function-like value.
+  // Prefer explicitly defining any function parameters and return type.
+  console.log(`Hello, ${getFriend().toUpperCase()}!`);
+}
+
+greet(() => 'Lazlo'); // Ok
+
+greet(() => ({ name: 'Nadya' })); // Should be a type error, but isn't
+```
+
+Instead of `Function`, it would have been more type safe to give `getFriend` the more precise type `() => string`.
+
+### Enforcing `unknown` for Caught Exceptions
+
+There is no `throws` keyword or other way to enforce types of exceptions in TypeScript[^no-throws], so their types can only be declared as `any` or `unknown`.
+Caught exceptions in `catch` blocks and `Promise` rejections are, by default, typed as `any` in TypeScript.
+TypeScript's [`useUnknownInCatchVariables`](https://www.typescriptlang.org/tsconfig/#useUnknownInCatchVariables) changes `catch` block exceptions to `unknown`, but no compiler option equivalent exists for `Promise` rejections.
+
+[`@typescript-eslint/use-unknown-in-catch-callback-variable`](https://typescript-eslint.io/rules/use-unknown-in-catch-callback-variable) enforces always using the unknown type for the parameter of a Promise rejection callback.
+
+For example, given the following code, TypeScript would not report a type error, but the lint rule would report:
+
+```ts
+function rejectWith(value: string) {
+  return Promise.reject(value);
+}
+
+rejectWith('Nandor').catch(error => {
+  //                       ~~~~~
+  // eslint(@typescript-eslint/use-unknown-in-catch-callback-variable):
+  // Prefer the safe `: unknown` for a `catch` callback variable.
+
+  console.log(error.message); // Should be a type error, but isn't
+});
+```
+
+Had the `error` been given a `: unknown` type, TypeScript would be able to report on `error.message` as being unsafe.
+
+## Banning Usage of Unsafe Types
+
+Once an `any` type exists in code, it is "infectious": it can turn the types of values based on it into more `any`s.
+typescript-eslint includes a collection of rules that flag code patterns which spread or are generally made less safe due to an `any`.
+
+### Banning Unsafe Arguments
```

Comment:
Going through each of the classic `any`-catchin `no-unsafe-*` rules is kind of tedious to read through 😕. I might try to find a way to condense these... not sure.